python/res/job_queue/job_queue_node.py

```diff
@@ -71,33 +75,40 @@ def is_running(self):
                 self.status == JobStatusType.JOB_QUEUE_RUNNING  or
                 self.status == JobStatusType.JOB_QUEUE_UNKNOWN) # dont stop monitoring if LSF commands are unavailable
     
-    def job_monitor(self, driver):
+    def job_monitor(self, driver, max_submit):
 
         self._submit(driver)
         self.update_status(driver)
 
         while self.is_running():
             time.sleep(1)
             self.update_status(driver)
+            if self.stopped_by_user:
+                self._kill(driver)
 
         if self.status == JobStatusType.JOB_QUEUE_DONE:
             self.run_done_callback()
         elif self.status == JobStatusType.JOB_QUEUE_EXIT:
-            self._set_status(JobStatusType.JOB_QUEUE_FAILED)
-            self.run_exit_callback()
+            if self.submit_attempt < max_submit:
+                self._set_status(JobStatusType.JOB_QUEUE_WAITING)
+                self.started = False
+            else:
+                self._set_status(JobStatusType.JOB_QUEUE_FAILED)
+                self.run_exit_callback()
         elif self.status == JobStatusType.JOB_QUEUE_WAITING:
             self.started = False
+        elif self.status == JobStatusType.JOB_QUEUE_IS_KILLED:
+            pass
```

Comment:
Should we add an:
```
else:
    raise AssertionError("Unexpected job status type after running job: {}".format(self.status))

Reply:
We can add this, but we should also clear out the job states we want to use